bioconda_utils/build.py

```diff
@@ -0,0 +1,351 @@
+import subprocess as sp
+from collections import defaultdict
+import os
+import logging
+import networkx as nx
+from . import utils
+from . import docker_utils
+from . import pkg_test
+from conda_build import api
+
+logger = logging.getLogger(__name__)
+
+
+def build(recipe,
+          recipe_folder,
+          env,
+          testonly=False,
+          mulled_test=True,
+          force=False,
+          channels=None,
+          docker_builder=None):
+    """
+    Build a single recipe for a single env
+
+    Parameters
+    ----------
+    recipe : str
+        Path to recipe
+
+    env : dict
+        Environment (typically a single yielded dictionary from EnvMatrix
+        instance)
+
+    testonly : bool
+        If True, skip building and instead run the test described in the
+        meta.yaml.
+
+    mulled_test : bool
+        Test the built package in a minimal docker container
+
+    force : bool
+        If True, the recipe will be built even if it already exists. Note that
+        typically you'd want to bump the build number rather than force
+        a build.
+
+    channels : list
+        Channels to include via the `--channel` argument to conda-build
+
+    docker_builder : docker_utils.RecipeBuilder object
+        Use this docker builder to build the recipe, copying over the built
+        recipe to the host's conda-bld directory.
+    """
+    env = dict(env)
+    logger.info(
+        "BIOCONDA BUILD START %s, env: %s",
+        recipe, ';'.join(['='.join(map(str, i)) for i in sorted(env.items())])
```

Comment:
I had tried that, but it turns out conda-build now expects integer python versions (27, 34, 35). For example the new env_matrix.yml over in bioconda-recipes (https://github.com/bioconda/bioconda-recipes/pull/2595) uses integers now. For flexibility in bioconda-utils I'm allowing int or str in CONDA_PY, so the extra map() make the `'='.join` happy. I could just dump the tuple straight to the log message, but it seemed like all the quotes, commas, and parentheses clutter things up.